src/data_source/storage/sql.rs

```diff
@@ -922,6 +922,7 @@ pub mod testing {
                 let path = format!("sqlite:{db_path}");
                 sqlx::sqlite::SqliteConnectOptions::from_str(&path)
                     .expect("invalid db path")
+                    .locking_mode(sqlx::sqlite::SqliteLockingMode::Exclusive)
```

Comment:
This seems...not great. IIUC this means the first time a connection locks the database for writing, it will never release that lock. That means any other connection will be unable to access the database until the first is closed, which means we get no parallelism from the connection pool. Would be good to understand if we really need this and why